src/main/java/org/alfresco/repo/management/JmxDumpUtil.java

```diff
@@ -164,9 +170,70 @@ private static void printMBeanInfo(MBeanServerConnection connection, ObjectName
                 attributes.put(OS_NAME, updateOSNameAttributeForLinux(osName));
             }
         }
+        if (objectName.getCanonicalName().equals("java.lang:type=Runtime"))
+        {
+            String[] commandInputs = (String[]) attributes.get(INPUT_ARGUMENTS);
+            if(commandInputs != null)
+            {
+                attributes.put(INPUT_ARGUMENTS, cleanPasswordsFromInputArguments(commandInputs));
+            }
+        }
         tabulate(JmxDumpUtil.NAME_HEADER, JmxDumpUtil.VALUE_HEADER, attributes, out, 0);
     }
 
+    /**
+     * If any of the string contain passwords, denote by "*password=" or "*pwd=" or "token="
+     * then they shall be replaces with JmxDumpUtil.PROTECTED_VALUE.
+     * @param commandInputs
+     * @return array of password redacted strings
+     */
+    public static String[] cleanPasswordsFromInputArguments(String[] commandInputs)
+    {
+        List<String> cleanInputs = new ArrayList<String>();
+        for (String input : commandInputs) 
+        {
+            input = cleanPasswordFromInputArgument(input);
+            cleanInputs.add(input);
+        }
+        
+        return cleanInputs.toArray(new String[commandInputs.length]);
+    }
+
+    /**
+     * Removes any characters following the words:
+     * password
+     * pwd
+     * token
+     * and replaces them with JmxDumpUtil.PROTECTED_VALUE
+     * 
+     * Example: 
+     * Input:   -Ddb.password=alfresco
+     * Output:  -Ddb.password=JmxDumpUtil.PROTECTED_VALUE
+     * 
+     * @param input
+     * @return password redacted string
+     */
+    public static String cleanPasswordFromInputArgument(String input)
+    {
+        // Regex to select the characters following the words password=, token= or pwd=
+        String pattern = "(?i)((?<=(password)=)|(?<=(token)=)|(?<=(pwd)=)).++";

```

Comment:
This can be extracted to a constant.

Reply:
Changed implementation of regex, it now has its own creation method: see lines 222 and lines 260-301